serve/filters/filter/tx.go

```diff
@@ -0,0 +1,101 @@
+package filter
+
+import (
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+)
+
+type Options struct {
+	Height    int64
+	Index     uint32
+	GasUsed   struct{ Min, Max int64 }
+	GasWanted struct{ Min, Max int64 }
+}
+
+// TxFilter holds a slice of transaction results.
+// It provides methods to manipulate and query the transactions.
+type TxFilter struct {
+	*baseFilter
+	txs  []*types.TxResult
+	opts Options
+}
+
+// NewTxFilter creates a new TxFilter object.
+func NewTxFilter(opts Options) *TxFilter {
+	return &TxFilter{
+		baseFilter: newBaseFilter(TxFilterType),
+		txs:        make([]*types.TxResult, 0),
+		opts:       opts,
+	}
+}
+
+// GetHashes iterates over all transactions in the filter and returns their hashes.
+func (tf *TxFilter) GetHashes() [][]byte {
+	tf.Lock()
+	defer tf.Unlock()
+
+	hashes := make([][]byte, 0, len(tf.txs))
+
+	for _, txr := range tf.txs {
+		var hash []byte
+
+		if txr != nil && txr.Tx != nil {
+			hash = txr.Tx.Hash()
+		}
+
+		hashes = append(hashes, hash)
+	}
+
+	return hashes
+}
+
+func (tf *TxFilter) UpdateWithTx(tx *types.TxResult) {
+	tf.Lock()
+	defer tf.Unlock()
+
+	tf.txs = append(tf.txs, tx)
+}
+
```

Comment:
The filter options need to be applied upon updating the filter with a tx result:
```go
func (tf *TxFilter) UpdateWithTx(tx *types.TxResult) {
	tf.Lock()
	defer tf.Unlock()

	tf.txs = checkOpts(tf.txs, tf.opts)
}
```

we can completely drop `Apply`